packages/pwa-kit-dev/src/utils/script-utils.ts

```diff
@@ -300,8 +353,26 @@ export const createBundle = async ({
                         existsSync(path.join(extendsTemplate, item))
                     )
                 }
+
+                // pwa-kit package versions are not listed as dependencies in package.json
+                // when a bundle is using template extensibility
+                const nestedPwaKitDependencies: {[key: string]: string} = {}
+                if (ccExtensibility.extends) {
+                    const dependencyTree = getAllProjectDependencyVersions()
+                    pwaKitDependencies.forEach((packageName) => {
+                        nestedPwaKitDependencies[packageName] = getLowestPackageVersion(
+                            packageName,
+                            dependencyTree
+                        )
+                    })
+                }
+
                 bundle_metadata = {
-                    dependencies: {...dependencies, ...devDependencies},
+                    dependencies: {
+                        ...dependencies,
+                        ...devDependencies,
+                        ...nestedPwaKitDependencies
+                    },
```

Comment:
is it possible for a customer to use template extensibility and also include a pwa-kit package listed under the dependencies in their package.json?

For example a package.json that looks like:
```
  ...
  "ccExtensibility": {
    "extends": "@salesforce/retail-react-app",
    "overridesDir": "overrides"
  },
  "devDependencies": {
    "@salesforce/retail-react-app": "1.0.1",
    "@salesforce/pwa-kit-dev": "3.2.0-dev",
    "@salesforce/pwa-kit-runtime": "3.2.0-dev"
  },
  ...
  ```

If yes, then I will update this to use the lowest pwa-kit version listed in `dependencies`, `devDependencies`, and `nestedPwaKitDependencies`

Reply:
oh thats right! Maybe I will create a function for gathering the bundle metadata so I can write a test for this isolated from the `createBundle` function